docs/plugins/kodiupdate.rst

```diff
@@ -16,6 +17,27 @@ which looks like this::
         user: kodi
         pwd: kodi
 
+With the basic configuration settings above, the ``kodiupdate`` plugin
+will notify Kodi to scan your entire music library after the import
+operation has completed. For large music libraries you may prefer to have
+Kodi only scan the directory where each album was imported. To enable
+this behavior, add `source` and `library` settings, like this:
+
+    kodi:
+        host: localhost
+        port: 8080
+        user: kodi
+        pwd: kodi
+        source: nfs://myserver.local/music/library/
+        library: /home/music/library/
```

Comment:
Could the plugin instead just use the global `directory` configuration value for beets? (Would this ever need to be different from that?)

Reply:
It's not well documented what the "library" config option *really* does, partly because I was trying to figure out how to describe it correctly. It has almost no bearing on the beets library path in a literal sense - it's used to remove the first part of the path from the album, which results in a string that is appended to the Kodi source. The beets library location could very well be different from the "library" configuration option. For example:

Kodi music source: nfs://192.168.3.5/home/public/music/
beets library: /home/public/music/library

After album import let's say the album path is "/home/public/music/library/Prince/Controversy".

In this scenario, "library" config should be "/home/public/music/". We strip that from the album path above, leaving: "library/Prince/Controversy". We then append the result to the Kodi music source to get "nfs://192.168.3.5/home/public/music/library/Prince/Controversy", which is a correct URL for Kodi.

The issue here is that the URL/path that Kodi works with may be fundamentally different from the beets library path.

Reply:
Hey, I'm really sorry for dropping the thread here a few months back! It sounds like we need to know _the Kodi equivalent_ of the beets `directory` path. Maybe we should call this `kodi_dir` or something similar to suggest that this needs to match a Kodi configuration parameter that determines _its_ base directory for music?

Reply:
No problem, been busy myself. 'kodi_dir' is fine, I just updated the code.